src/main/java/com/exadel/discountwebapp/location/service/LocationService.java

```diff
@@ -49,14 +54,20 @@ public Location findEntityById(Long id) {
     @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public List<CityResponseVO> findAllCitiesByCountryCode(String countryCode) {
         List<Location> locations = locationRepository.findAllByCountry_CountryCode(countryCode);
-        List<CityResponseVO> response = new ArrayList<>();
-        locations.forEach(entity -> response.add(cityMapper.toVO(entity)));
-        return response;
+        List<CityResponseVO> allItems = new ArrayList<>();
+        locations.forEach(entity -> allItems.add(cityMapper.toVO(entity)));
+        return allItems.stream()
+                .filter(el -> el.getCity() != null && !el.getCity().equals("")).distinct()
```

Comment:
it's better use StringUtils.isNotBlank( ... ) from the commons lib

Reply:
Used, thanks